### PR TITLE
Simplification and clean up of Gradle files

### PR DESCRIPTION
**Pull request by meenbeese**
I tried to clean up our Gradle files a bit by:

- Implementing some linting suggestions
- Moving some things around for clarity
- Replacing deprecated methods with new ones
- Transferring all repo declarations to the settings file

P.S. Gradle sync works fine but please test for yourself if the app compiles. I can't test because of a
`java.lang.UnsatisfiedLinkError: dlopen failed: library "libzxing.so" not found` error.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -2,11 +2,12 @@ apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
 
 android {
-	namespace 'de.markusfisch.android.binaryeye'
```
Comment (markusfisch) [inline app/build.gradle:5]: Why did you put this line at the end of the block? For me, defining a namespace is something I like to do next. Are there any reasons I haven't thought of?

File: app/build.gradle [hunk 2]
```diff
@@ -2,11 +2,12 @@ apply plugin: 'com.android.application'
...
-	compileSdkVersion sdk_version
```
Comment (markusfisch) [inline app/build.gradle:6]: Why did you remove this line?

File: app/build.gradle [hunk 3]
```diff
@@ -2,11 +2,12 @@ apply plugin: 'com.android.application'
...
 	defaultConfig {
-		minSdkVersion 9
```
Comment (markusfisch) [inline app/build.gradle:8]: Because the actual value, "9", is used just once, in this place, I don't see much value in defining a global symbol for it.

File: build.gradle [hunk 4]
```diff
@@ -18,19 +19,15 @@ buildscript {
 }
 
 allprojects {
-	repositories {
-		google()
-		mavenCentral()
-		maven { url 'https://jitpack.io' }
-	}
```
Comment (markusfisch) [inline build.gradle:25]: Why move this block in `settings.gradle`? 🤔 And have it there two times for `pluginManagement` and `dependencyResolutionManagement`?

**Comment by markusfisch** [comment]
Thanks for the contribution! I'm just wondering about what's the rationale behind some of these changes 🤔 Please check my single line comments.

> I can't test because of a `java.lang.UnsatisfiedLinkError: dlopen failed: library "libzxing.so" not found` error.

Did you maybe try to run the app on x86 hardware? `libzxing.so` is the name of the native library from the `zxing-cpp` dependency, and until very recently this was available for the arm architecture only. With version 2.2.x (of `zxing-cpp`), x86 is supported too, so `master` should run just fine.

File: app/build.gradle (hunk 1 above)
Comment (meenbeese) [reply app/build.gradle:5]: Your argument makes sense. My personal preference is at the bottom, but I will move it up.

File: app/build.gradle (hunk 2 above)
Comment (meenbeese) [reply app/build.gradle:6]: I had moved it into the defaultConfig block, reverting that change now.

File: app/build.gradle (hunk 3 above)
Comment (meenbeese) [reply app/build.gradle:8]: This seems reasonable as well, I'm reverting the change.

File: app/build.gradle [hunk 5]
```diff
@@ -2,11 +2,12 @@ apply plugin: 'com.android.application'
...
-		targetSdkVersion sdk_version
+		applicationId 'de.markusfisch.android.binaryeye'
```
[the remark this answers is not in the document]
Comment (meenbeese) [reply app/build.gradle:6]: This is how it is done in practically all Android apps, I also don't see any harm in keeping it.

File: build.gradle (hunk 4 above)
Comment (meenbeese) [reply build.gradle:25]: This is the modern way to do it in Android apps. In fact, the top-level build file should only define plugins and nothing else. Defining dependencies is the job of `settings.gradle`.

**Comment by meenbeese** [comment]
> Did you maybe try to run the app on x86 hardware? `libzxing.so` is the name of the native library from the `zxing-cpp` dependency, and until very recently this was available for the arm architecture only. With version 2.2.x (of `zxing-cpp`), x86 is supported too, so `master` should run just fine.

Yes, I am building the app on x86 Windows. Yet, I don't have any experience with C/C++ so I don't think I will be able to compile any native libraries on my machine. I looked up `zxing-cpp` but couldn't find any easy installation instructions.

File: app/build.gradle (hunk 5 above)
[the remark this answers is not in the document]
Comment (markusfisch) [reply app/build.gradle:9]: Well, it's in practically all Android apps because Android Studio creates new projects with it, and probably because nobody knows it's optional.

Now, I'm not trying to convince you if you like to specify it. I'm just trying to make clear that I like to limit things to what is necessary. For my taste, I like it when you can't leave anything out - and not when you can't add anything in.

So I do consider unnecessary redundancies to be at least a potential problem. Now that every app must contain a `namespace`, and this should reasonably always correspond to the `applicationId`, duplicating this String definitely feels wrong to me. Whenever I need a string in two different places, I use a symbol. Of course, you could do the same here - but what for, if `applicationId` is not necessary anyway?

Sorry for being picky about these (completely unimportant) things. You don't need to take this out, but I will likely remove it after the merge. Just so you know 😉

File: app/build.gradle [hunk 6]
```diff
@@ -62,15 +62,21 @@ android {
...
 	implementation "com.android.support:appcompat-v7:$support_version"
 	implementation "com.android.support:design:$support_version"
 	implementation "com.android.support:preference-v7:$support_version"
 	implementation "com.android.support:preference-v14:$support_version"
+
+	// External
 	implementation 'com.github.markusfisch:CameraView:1.9.2'
 	implementation 'com.github.markusfisch:ScalingImageView:1.4.1'
-	implementation 'com.github.markusfisch:zxing-cpp:v2.2.0.1'
+	implementation 'com.github.markusfisch:zxing-cpp:v2.1.0.2'
```
Comment (markusfisch) [inline app/build.gradle:78]: That's the wrong version 😬 If you use `v2.2.0.1`, it should compile and run on your system, too. This is what is specified on `master`.

File: build.gradle (hunk 4 above)
Comment (markusfisch) [reply build.gradle:25]: Ah, okay, I see 👍

**Comment by markusfisch** [comment]
Thanks for the quick response and your patience. If you fix the wrong version number of `zxing-cpp` (see the single line comment), I will merge it.

File: app/build.gradle (hunk 6 above)
Comment (meenbeese) [reply app/build.gradle:78]: I made a small mistake while rebasing because I manually checked what was conflicting and updated them 😁

File: app/build.gradle (hunk 5 above)
[the remark this answers is not in the document]
Comment (meenbeese) [reply app/build.gradle:9]: Thanks for the detailed explanation and letting me know that it is optional. I totally understand your reasoning and I respect people's different coding styles. If you are going to remove it after the merge anyway, I see no reason on pushing it in the first place. I have now removed the `applicationId` from this PR.

**Comment by meenbeese** [comment]
> Thanks for the quick response and your patience. If you fix the wrong version number of `zxing-cpp` (see the single line comment), I will merge it.

I fixed the wrong version number. Should I also squash my commits, because some projects prefer that? As well, updating the `zxing-cpp` library fixed my problem and I was able to compile the program. I can confirm that the app works correctly when tested on Pixel 6a API 34.

**Comment by markusfisch** [comment]
Thanks a again for your work on this! 👍
